spec/lib/percy/client/connection_spec.rb

```diff
@@ -1,114 +1,143 @@
 RSpec.describe Percy::Client::Connection do
+  let(:user_agent) do
+    "Percy/#{api_version} #{client_info} percy-client/#{Percy::Client::VERSION} "\
+    "(#{environment_info}; ruby/#{ruby_version}; #{ci_name})"
+  end
+  let(:content_type) { 'application/vnd.api+json' }
+  let(:api_version) { 'v1' }
+  let(:ruby_version) { '2.2.6p396' }
+  let(:client_info) { 'percy-capybara/3.1.0' }
+  let(:environment_info) { 'Rails/4.2.1' }
+  let(:ci_name) { 'buildkite' }
+  let(:uri) { "#{Percy.config.api_url}/test" }
+
   describe '#connection' do
     it 'disables cookies on faraday httpclient adapter' do
       expect(Percy.client.connection.builder.app.client.cookie_manager).to be_nil
     end
   end
+
+  shared_examples_for 'a connection that sets headers with HTTP method' do |http_method|
+    it 'sets headers' do
+      stub_request(http_method, uri)
+        .with(headers: {'User-Agent' => user_agent, 'Content-Type' => content_type})
+        .to_return(body: {foo: true}.to_json)
+
+      expect(Percy.client).to receive(:_api_version).and_return(api_version)
+      expect(Percy.client).to receive(:_ruby_version).and_return(ruby_version)
+
+      expect(Percy.client).to receive(:client_info).and_return(client_info)
+      expect(Percy.client).to receive(:environment_info).and_return(environment_info)
+
+      expect(Percy::Client::Environment).to receive(:current_ci).and_return(ci_name)
+
+      expect(response).to eq('foo' => true)
+    end
+  end
+
   describe '#get' do
+    subject(:response) { Percy.client.get(uri) }
```

Comment:
Reading this confused me for a minute, the fact that `:response` is actually executing the `get` method itself, and it's lazy-loaded so when you're doing `expect(response).to eq('foo' => true)` it's actually first calling `Percy.client.get(uri)` and then following through on the expect. Definitely a tradeoff though, this is a nice way to DRY up the tests a bit by removing that duplicate line, but wanted to mention that it might be somewhat harder to read through.